cmd/dp-dataset-exporter/main.go

```diff
@@ -198,3 +216,12 @@ func logIfError(err error) {
 		log.Error(err, nil)
 	}
 }
+
+func updateInitialisation(err error, service map[string]bool, name string) {
```

Comment:
Question: Is this only ever used on start up sequentially? I'm wondering if there is a need to add a `lock` here to prevent concurrent modification issues.

Reply:
It is only called on startup sequentially; I can't see it being reused in a concurrent manner unless the code changes to initialise clients asynchronously and if it does occur I would think to add a lock on it at that point. I ran `go run -race main.go` and there wasn't a race condition on the map.